### PR TITLE
Make sure status is never undefined in onProgress server callback #5

### PR DESCRIPTION
**Pull request by jviotti**

### CODE REVIEW

File: lib/server.coffee [hunk 1]
```diff
@@ -111,7 +113,6 @@ exports.request = (options = {}, outerCallback, onProgress) ->
 				progress(request(options))
 					.on('progress', onProgress)
 					.on('error', outerCallback)
-					.on('end', onProgress)
```
Comment (Page-) [inline lib/server.coffee]: Is it worth it/useful to still expose the `end` event?

**Comment by Page-** [comment]
Is there anything that can be done about the docs update noise?  It makes these PRs a lot harder to look at since there's a huge number of trivially changed files

**Comment by jviotti** [comment]
@Page- I agree as well. JSDoc seems to update every file at every change (it's frustrating when developing as well, as it makes very difficult to commit part of my changes). It seems JSDoc updated the footer of every file with the new date. I'll investigate if I can prevent that from happening.

File: lib/server.coffee (hunk 1 above)
Comment (jviotti) [inline lib/server.coffee]: @Page- I've did some testing, at `close` handles all cases. Having the `end` listener call `onProgress` results in `onProgress` having a `null` state argument, and calling it with `outerCallback` made such callback be called twice (by `end` and `close`).

Comment (Page-) [inline lib/server.coffee]: Doesn't `close` get called on `error` as well, which would result in a double call there?

Comment (jviotti) [inline lib/server.coffee]: After some experimentation, `end` is not even being called by `node-request-progress`. Also by checking their example, they use `close` to determine the final callback, so I think we're good like it is (I wasn't able to reproduce a case where `outerCallback` was called twice neither).

**Pull request merged**